src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -298,7 +298,9 @@ public static function createFromGlobals()
     {
         $request = self::createRequestFromFactory($_GET, $_POST, [], $_COOKIE, $_FILES, $_SERVER);
 
-        if (0 === strpos($request->headers->get('CONTENT_TYPE'), 'application/x-www-form-urlencoded')
+        if ($_POST) {
+            $request->request = new InputBag($_POST);
```

Comment:
Most of the developers using symfony that will create a form using `form` component, let's say a login form, will get the data back as 

```
`login` => [
    `username` => `user`,
    `password` =>  `password`,
];
```

So I don't see this InputBag here serving the majority..

Reply:
The form component is already strict about strings vs arrays in values and knows how to deal with InputBag.

Reply:
Yes the form component, as a component knows how to deal with the InputBag.. my point is that users that will build a form with the component... they will never use this:

`$request->request->get('login');`

What use cases/bugs are you trying to cover by using for the `request` attribute the InputBag?

> This means many apps just fail with a 500 when adding [] in the query string.

All this started with the scenario that someone access a query parameter and thinking that it is string etc etc.. ok.. although I disagree with the notion of it.. it is a UseCase. I don't see how developers will be protected here... 

All I see is that I have to do this:

```php
        $login = $request->request->get('login');
        return \is_array($login) && isset($login['username']) && isset($login['password']) && isset($login['_token']);
```

to:

```php
        $request = $request->request->all();
        if (!isset($request['login'])) {
            return false;
        }

        $login = $request['login'];
        return \is_array($login) && isset($login['username']) && isset($login['password']) && isset($login['_token']);
```

it's not the code that I have to write.. it's that `ParameterBag::get` is totally useless.. 

Hope I am explaining in the best way.. if not.. maybe we should move on.. there might be more important things to care..

Reply:
Your "to" should be:
```php
$request = $request->request->all('login');
return isset($login['username']) && isset($login['password']) && isset($login['_token']);
```

Then you'll let a `BadRequestException` bubble down and become a 400 as it should if you expect `login` to be an array. This "after" is much better than the "before" actually, with a richer behavior and more semantic code.

Reply:
> Then you'll let a BadRequestException bubble down and become a 400 as it should if you expect login to be an array.

I can't.. various technical reasons that are out of the topic of this discussion...

Still this remains:

> it's not the code that I have to write.. it's that ParameterBag::get is totally useless..